aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/data/AWSCognitoLegacyCredentialStore.kt

```diff
@@ -149,12 +158,32 @@ class AWSCognitoLegacyCredentialStore(
 
     private fun retrieveSignedInData(): SignedInData? {
         val keys = getTokenKeys()
+        val cognitoUserPoolTokens = retrieveCognitoUserPoolTokens(keys) ?: return null
         val username = keys[APP_LAST_AUTH_USER]?.let { tokensKeyValue.get(it) }
         val deviceMetaData = retrieveDeviceMetadata(username)
-        val cognitoUserPoolTokens = retrieveCognitoUserPoolTokens(keys)
+        val signInMethod = retrieveUserPoolSignInMethod()
+            ?: SignInMethod.ApiBased(SignInMethod.ApiBased.AuthType.USER_SRP_AUTH)
+        val tokenUserId =
+            try {
+                cognitoUserPoolTokens.accessToken?.let { SessionHelper.getUserSub(it) } ?: ""
+            } catch (e: Exception) {
+                ""
+            }
+        val tokenUsername =
+            try {
+                cognitoUserPoolTokens.accessToken?.let { SessionHelper.getUsername(it) } ?: ""
+            } catch (e: Exception) {
+                ""
+            }
 
-        return if (cognitoUserPoolTokens == null) null
-        else SignedInData("", username ?: "", Date(0), SignInMethod.SRP, deviceMetaData, cognitoUserPoolTokens)
+        return SignedInData(
```

Comment:
TODO: Follow up coming soon in federation to identity pool API where signed in data will be null.